tests/Data/ProgrammesDb/EntityRepository/PromotionRepository/FindActivePromotionsByContextTest.php

```diff
@@ -10,25 +11,95 @@
  */
 class FindActivePromotionsByContextTest extends AbstractDatabaseTest
 {
-    public function testActivePromotionsAreReceived()
+    /**
+     * @var PromotionRepository
+     */
+    private $promotionRepository;
+
+    public function setUp()
     {
+        parent::setUp();
+
         $this->loadFixtures(['PromotionsFixture']);
-        $promotionRepository = $this->getRepository('ProgrammesPagesService:Promotion');
+        $this->promotionRepository = $this->getRepository('ProgrammesPagesService:Promotion');
+    }
+
+    public function testActiveSuperpromotionsAreReceivedWithSuperPromotionsForEpisode()
+    {
+        $coreEntityRepo = $this->getEntityManager()->getRepository('ProgrammesPagesService:CoreEntity');
+
+        $series = $coreEntityRepo->findByPid('b00syxx6', 'Episode'); // series 1/episode 1
+        $seriesDbAncestryIds = array_filter(explode(',', $series['ancestry']));
+
+        $dbPromotions = $this->promotionRepository->findActivePromotionsByContext($seriesDbAncestryIds, new DateTimeImmutable(), 300, 0);
+
+        // It doesn't fetch promotions from the series because the ones in series 1 are not super promotions.
+        // But the one in brand (p000000h), which is superpromotions, is fetched
+        $this->assertEquals(['p000004h', 'p000000h'], array_column($dbPromotions, 'pid'));
+        $this->assertEquals(
+            ['episode', 'series', 'brand'],
+            $this->getParentTypesRecursively($dbPromotions[0]['promotionOfCoreEntity'])
+        );
 
-        $dbPromotions = $promotionRepository->findActivePromotionsByContext(1, new DateTimeImmutable(), 300, 0);
+        $this->assertCount(3, $this->getDbQueries());
+    }
 
-        $this->assertEquals(['p000000h', 'p000001h', 'p000004h'], array_column($dbPromotions, 'pid'));
+    public function testActiveSuperpromotionsAreReceivedWithSuperPromotionsForSeries()
+    {
+        $coreEntityRepo = $this->getEntityManager()->getRepository('ProgrammesPagesService:CoreEntity');
+
+        $series = $coreEntityRepo->findByPid('b00swyx1', 'Series'); // series 1/episode 1
+        $seriesDbAncestryIds = array_filter(explode(',', $series['ancestry']));
```

Comment:
`$this->getAncestryFromPersistentIdentifier('b00swyx1', 'CoreEntity')` is a utility method provided by the base class to do this. It's particularly useful because it doesn't count the query when you call `getDbQueries`.